craft_store/publisher/_response.py

```diff
@@ -0,0 +1,71 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Response models for the publisher gateway."""
+
+from typing import TypedDict
+
+from typing_extensions import NotRequired
+
+PublisherMetadata = TypedDict(
```

Comment:
don't we loose runtime benefits by going down this path?

Reply:
I don't see how. At runtime, these are just standard dicts. We can wrap these imports behind TYPE_CHECKING if you'd like, but I think the amount of time needed to load this is pretty insignificant.

Reply:
ah, I mean in comparison to pydantic models where the data is validated

Reply:
Oh - yes, in that case we do, but if the store returns valid JSON that's still not what we expect we're in a world of hurt either way. I thought you were trying to remove the pydantic dependency here though?